rpcq/_base.py

```diff
@@ -42,68 +43,27 @@ class UnknownMessageType(Exception):
     """Raised when trying to decode an unknown message type."""
 
 
-class Message(object):
+class Message:
     """
     Base class for messages.
     """
 
-    def __repr__(self):
-        return "{}({})".format(
-            self.__class__.__name__,
-            ", ".join("{}={}".format(k, repr_value(v))
-                      for k, v in sorted(self.asdict().items(), key=lambda kv: kv[0])))
-
-    def __getitem__(self, item):
-        return self.asdict()[item]
-
-    def items(self):
-        return self.asdict().items()
-
-    def get(self, key, default):
-        return self.asdict().get(key, default)
-
-    def __eq__(self, other):
-        return type(self) == type(other) and self.astuple() == other.astuple()
-
-    def __copy__(self):
-        return self.__class__(**self.asdict())
-
-    def __deepcopy__(self, memo):
-        ret = self.__class__(**deepcopy(self.asdict(), memo=memo))
-        memo[id(self)] = ret
-        return ret
-
-    def copy(self, **kwargs):
-        """
-        Create a copy with (optionally) substituted values.
-
-        :param kwargs: Any fields that should be substituted.
-        :return: A new message object of the same class.
-        """
-        values = self.asdict()
-        values.update(kwargs)
-        return self.__class__(**values)
-
     def asdict(self):
         """
         Create a dictionary ``{fieldname1: fieldvalue1, ...}`` of the Message object.
 
         :return: A dictionary representation of the message.
         :rtype: Dict[str,Any]
         """
-        raise NotImplementedError(self.__class__.__name__)
+        return self.__dict__.copy()
 
-    def astuple(self):
-        """
-        Create a tuple ``{fieldvalue1, ...}`` of the Message object.
-
-        :return: A tuple representation of the message.
-        :rtype: Tuple[Any]
-        """
```

Comment:
@ecp-rigetti, @stevenheidel Can we at least explicitly canonicalize the order?
I think doing something like
```
from dataclasses import fields

class Message:
    [...]

    def astuple(self):
        return tuple(getattr(self, f.name) for f in fields(self))
```

Reply:
I see. Sure. (For onlookers / reference, `dataclasses.fields()` doesn't do anything like sorting, but it is stored at the class-definition level, so it's at least consistent across a particular run: https://github.com/python/cpython/blob/3.7/Lib/dataclasses.py#L873 .)